### PR TITLE
Implement full screen mode toggle

### PR DESCRIPTION
**Pull request by weeman1337**
Based on https://github.com/nordeck/matrix-neoboard/pull/104

fullscreen-mode

**:heavy_check_mark: Checklist**

- [x] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [x] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: src/lib/testUtils/documentTestUtils.tsx [hunk 1]
```diff
@@ -258,3 +258,36 @@ export function mockTextElement(shape: Partial<ShapeElement> = {}): Element {
...
+
+/**
+ * Mock the fullscreen API {@link https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API}
+ *
+ * Set document.fullscreenElement to null.
+ * Mock document.exitFullscreen() to reset document.fullscreenElement to null.
+ * Also mock document.documentElement.requestFullscreen() to set document.fullscreenElement to an empty object.
+ * Both mocked functions dispatch a "fullscreenchange" event on document and return a resolved Promise.
+ */
+export function mockFullscreenApi(): void {
```
Comment (weeman1337) [inline src/lib/testUtils/documentTestUtils.tsx:270]: Taks for me: Have another look if the API can be mocked in a better way.

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -99,6 +99,11 @@
     "deleteElement": "Delete",
     "title": "Element"
   },
+  "fullscreenModeBar": {
+    "disableFullscreenMode": "Disable fullscreen mode",
+    "enableFullscreenMode": "Enable fullscreen mode",
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply public/locales/en/translation.json:104]: Yes, you are right. Although it should be „Fullscreen mode on/off“ to be consistent with the other language. Let's clarify with the PO.

File: src/components/FullscreenModeBar/ToggleFullscreenButton.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+    <ToolbarToggle
+      inputProps={{ 'aria-label': fullscreenModeButtonTitle }}
+      checked={isFullscreen}
+      onClick={toggleFullscreenMode}
+      icon={<FullscreenIcon />}
+      placement="bottom"
+      checkedIcon={<FullscreenExitIcon />}
+    />
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/components/FullscreenModeBar/ToggleFullscreenButton.tsx:44]: Good catch! According to the designs it should not be highlighted.

File: src/components/Layout/Layout.tsx [hunk 4]
```diff
@@ -28,6 +28,7 @@ import { BoardBar } from '../BoardBar';
 import { CollaborationBar } from '../CollaborationBar';
 import { DeveloperTools } from '../DeveloperTools/DeveloperTools';
 import { ElementOverridesProvider } from '../ElementOverridesProvider';
+import { FullscreenModeBar } from '../FullscreenModeBar/FullscreenModeBar';
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/components/Layout/Layout.tsx:31]: Module created :heavy_check_mark:

File: src/components/FullscreenModeBar/ToggleFullscreenButton.test.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    checked: false,
+  });
+}
+
+function getDisableFullscreenModeButton() {
+  return screen.getByRole('checkbox', {
+    name: 'Disable fullscreen mode',
+    checked: true,
+  });
+}
```
Comment (maheichyk) [inline src/components/FullscreenModeBar/ToggleFullscreenButton.test.tsx:82]: Let's inline the usages of these functions to use `screen.getByRole` directly in the code. Each of these functions is used only once, so no big value to introduce a separate function there. In many other tests we don't introduce extra functions for similar cases.

**Review by maheichyk — CHANGES_REQUESTED** [review]

File: src/components/FullscreenModeBar/ToggleFullscreenButton.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/components/FullscreenModeBar/ToggleFullscreenButton.tsx:44]: Changed to a button.

File: public/locales/en/translation.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (weeman1337) [reply public/locales/en/translation.json:104]: Updated

File: src/lib/testUtils/documentTestUtils.tsx (hunk 1 above)
Comment (weeman1337) [reply src/lib/testUtils/documentTestUtils.tsx:270]: Mocking this API seems to be a strange thing. I've tried everything from `Object.defineProperty(document, 'exitFullscreen'…` to extending the `document` prototype. Neither has worked for me :shrug:  Open for ideas.

Comment (charlynguyen) [reply src/lib/testUtils/documentTestUtils.tsx:270]: Let's improve it later on.

**Review by charlynguyen — APPROVED** [review]
LGTM (mocking can be improved later on 🙂)

File: src/components/FullscreenModeBar/ToggleFullscreenButton.test.tsx (hunk 5 above)
Comment (weeman1337) [reply src/components/FullscreenModeBar/ToggleFullscreenButton.test.tsx]: Done

**Review by maheichyk — APPROVED** [review]

**Pull request merged**